src/App.tsx

```diff
@@ -1,24 +1,220 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  FormEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  addTodo, deleteTodo, getTodos, updateTodo,
+} from './api/todos';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+import { TodoItem } from './components/TodoItem';
+import { TodoList } from './components/TodoList';
+import { ToggleButton } from './components/ToggleButton';
+import { ErrorTypes } from './types/ErrorTypes';
+import { FilterCases } from './types/FilterCases';
+import { Todo, TodoToSend } from './types/Todo';
+import { filterByStatus, generateError } from './utils/helper';
 
-const USER_ID = 0;
+const USER_ID = 6683;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState(FilterCases.All);
+  const [error, setError] = useState(ErrorTypes.None);
+  const [inputValue, setInputValue] = useState('');
+  const [isAddingProceeding, setIsAddingProceeding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<null | Todo>(null);
+  const [isUpdated, setIsUpdated] = useState(false);
+
+  const fetchTodos = async () => {
+    const todosFromServer = await getTodos(USER_ID);
+
+    setTodos(todosFromServer);
+  };
+
+  const handleTodoDelete = async (id: number) => {
+    try {
+      await deleteTodo(id);
+
+      setTodos(current => current.filter(todo => todo.id !== id));
+    } catch {
+      generateError(ErrorTypes.DeleteTodoError, setError);
+    }
+  };
+
+  const clearCompletedTodos = () => {
+    const arrayOfPromises:Promise<void>[] = [];
+    const completedTodos = todos
+      .filter(({ completed }) => completed);
+
+    completedTodos.forEach(todo => {
+      arrayOfPromises.push(handleTodoDelete(todo.id));
+    });
+    Promise.all(arrayOfPromises);
+  };
+
+  useEffect(() => {
+    fetchTodos();
+  }, [isUpdated]);
+
+  const handleFilterUpdate = useCallback((newFilter: FilterCases) => {
+    setFilterType(newFilter);
+  }, []);
+
+  const handleNotificationClose = useCallback(() => {
+    setError(ErrorTypes.None);
+  }, []);
+
+  const handleTodoAdd = useCallback((data: TodoToSend) => {
+    const todo = addTodo(data);
+
+    setTempTodo({
+      ...data,
+      id: 0,
```

Comment:
wait instance form back-end with added todo to database and add it to the state with right id